### PR TITLE
OpenCV Cascade Integration

### PR DESCRIPTION
**Pull request by JordanCheney**
@bklare @sklum @jklontz I'm opening this as a checklist for cascade integration so you all can stay up to date on the progress-

### Backend
- [x] Integrate training code using the filesystem as an intermediary (i.e write to disk with OpenBR and then read with the OpenCV train code)
- [x] Remove the file system middle man so images can be preprocessed in OpenBR and then passed directly to train code
- [x] Incorporate representations instead of OpenCV features into the backend
- [x] Fit the backend code into the Classification / SlidingWindow paradigm
- [x] Convert to OpenBR I/O

### Frontend
- [x] Implement the sliding window front end
- [x] Incorporate representations instead OpenCV features
- [x] Add to sliding window / classification
- [x] Convert to OpenBR I/O

### CODE REVIEW

**Comment by jklontz** [comment]
:+1:

**Comment by sklum** [comment]
Sounds good @JordanCheney - looking forward to the progression.

**Comment by JordanCheney** [comment]
Updated the checklist to reflect an easier path

**Comment by sklum** [comment]
@JordanCheney IMO, this can and should be merged ASAP.

File: openbr/plugins/representation/square.cpp [hunk 1]
```diff
@@ -0,0 +1 @@
+
```
Comment (jklontz) [inline openbr/plugins/representation/square.cpp]: extraneous file?

File: openbr/core/boost.h [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    float minHitRate;
+    float maxFalseAlarm;
+
+    CascadeBoostParams();
+    CascadeBoostParams(int _boostType, float _minHitRate, float _maxFalseAlarm,
+                       double _weightTrimRate, int _maxDepth, int _maxWeakCount);
+    virtual ~CascadeBoostParams() {}
+};
+
+struct CascadeBoostTrainData : CvDTreeTrainData
```
Comment (jklontz) [inline openbr/core/boost.h]: Seems to me that `CascadeBoostTrainData` and `CascadeBoostTree` need not be exposed in this header file as they are only used in `boost.cpp`.

File: openbr/plugins/representation/square.cpp (hunk 1 above)
Comment (JordanCheney) [inline openbr/plugins/representation/square.cpp]: Yep, removed.

File: openbr/core/boost.h (hunk 2 above)
Comment (JordanCheney) [inline openbr/core/boost.h]: ```CascadeBoostTree``` is used by ```BoostedForest```. ```CascadeBoostTrainData``` is moved to ```boost.cpp```

**Pull request merged**